src/visitors/visitor_utils.cpp

```diff
@@ -215,4 +227,28 @@ std::string to_json(const ast::Ast& node, bool compact, bool expand, bool add_nm
     return stream.str();
 }
 
+std::pair<std::string, std::unordered_set<std::string>> statement_dependencies(
+    const std::shared_ptr<ast::Expression>& lhs,
+    const std::shared_ptr<ast::Expression>& rhs) {
+    std::string key;
+    std::unordered_set<std::string> out;
+
+    if (!lhs->is_var_name()) {
+        return {key, out};
+    }
+
+    key = to_nmodl(lhs);
```

Comment:
mmm yes, good. Thx

Reply:
`ast::Expression` does not have a `get_name` function. I could use `get_var_name` but I would miss the `[0]` for the vectors. Since I did not understand your idea fully, I dunno how to correct the code. What is the node that should be used. Notice that this function should work for:

- diffEqExpression
- linEquation
- nonLinEquation
- binary expression